document/concepts/overview.rst

```diff
@@ -1,6 +1,91 @@
 Overview
 --------
 
-.. todo::
+WebAssembly encodes a low-level, assembly-like programming language.
+This language is structured around the following main concepts.
 
-   Describe
+* **Values**.
+  WebAssembly provides only four basic *value types*.
+  These are integers and :ref:`IEEE-754 floating point <http://ieeexplore.ieee.org/document/4610935/>` numbers,
+  each in 32 and 64 bit width.
+  32 bit integers also function as Booleans and as memory addresses.
+  The usual basic operations on these types are available,
+  including the full matrix of conversions between them.
+  There is no distinction between signed and unsigned integer types.
+  Instead, different signedness is handled by different operations,
+  selecting either unsigned or 2’s complement signed interpretation.
+
+* **Instructions**.
+  The computational model of WebAssembly is based on a *stack machine*.
+  Code consists of sequences of *instructions* that are executed in order.
+  They manipulate values on an implicit *operand stack*. [#stackmachine]_
+  Instructions fall into two main categories.
+  *Simple* instructions perform basic operations on data.
+  They pop argument values from the operand stack and push result values back to it.
+  *Control* instructions alter control flow.
+  They are either sources or targets of *branches*.
+  Control flow is *structured*, i.e., it is expressed with well-nested constructs such as blocks, loops, and conditionals.
+  Branches can only target such constructs.
+
+* **Traps**.
+  Some instructions may produce a *trap* under some conditions,
+  which immediately aborts excecution.
+  Traps cannot be handled by WebAssembly code,
+  but are reported to the outside environment,
+  where they typically can be caught.
+
+* **Functions**.
+  Code is organized into separate *functions*.
+  Each function takes a sequence of values as parameters
+  and returns zero or one values as a result.
+  Functions can call each other, including recursively.
+  The call stack is implicit and cannot be accessed directly.
+  Nor is the address of a function accessible.
+  Functions may also declare mutable *local variables* that are usable as virtual registers.
+
+* **Tables**.
+  A *table* is an array of opaque values of a particular *element type*.
+  It allows programs to select such values indirectly through a dynamic index operand.
+  Currently, the only available element type is an untyped function reference.
+  Thereby, code can perform an indirect call to a function indexed through a table.
```

Comment:
s/Thereby, code can perform.../A program can call functions indirectly by calling into one of these opaque tables with a dynamically computed integer index.